scripts/polygon_cleanup_tracks.py

```diff
@@ -0,0 +1,127 @@
+    return rings
+
+
+def main(arguments):
+    # initialize logger
+    logger = initialize_logging(arguments.log_file)
+    # Create the GIS
+    logger.info("Authenticating...")
+    # First step is to get authenticate and get a valid token
+    gis = GIS(arguments.org_url,
+              username=arguments.username,
+              password=arguments.password,
+              verify_cert=not arguments.skip_ssl_verification)
+    
+    logger.info("Getting location tracking service")
+    try:
+        tracks_layer = gis.admin.location_tracking.tracks_layer
+    except Exception as e:
+        logger.info(e)
+        logger.info("Getting location tracking service failed - check that you are an admin and that location tracking is enabled for your organization")
+        sys.exit(0)
+    
+    logger.info("Getting polygon layer")
+    try:
+        layer = FeatureLayer(url=args.layer_url, gis=gis)
+        json = layer._lyr_json
+    except Exception as e:
+        logger.info(e)
+        logger.info("Layer could not be found based on given input. Please check your parameters again. Exiting the script")
+        sys.exit(0)
+    
+    features = layer.query(where=args.where, out_sr=4326).features
+    if len(features) > 0:
+        geometries = [feature.geometry for feature in features]
+        logger.info("Unifying geometry data")
+        union_geometry = geometry.union(spatial_ref=4326,geometries=geometries,gis=gis)
+        if not strtobool(args.inside):
+            union_geometry['rings'] = inverse_geometry(union_geometry['rings'])
+        intersect_filter = geometry.filters.intersects(union_geometry, sr=4326)
+        logger.info("Querying features")
+        x = tracks_layer.delete_features(geometry_filter=intersect_filter)
+        logger.info("Deleted: " + str(x))
+        logger.info("Completed!")
+    
+    
+if __name__ == "__main__":
+    # Get all of the commandline arguments
+    parser = argparse.ArgumentParser(
+        "This sample allows cleanup of track points from a tracks layer based on a spatial relationship to polygon geometry")
+    parser.add_argument('-u', dest='username', help="The username to authenticate with", required=True)
+    parser.add_argument('-p', dest='password', help="The password to authenticate with", required=True)
+    parser.add_argument('-org', dest='org_url', help="The url of the org/portal to use", required=True)
+    # Parameters for tracker
+    parser.add_argument('-layer-url', dest='layer_url',
+                        help="The feature service URL for your layer with the geometry you want to use to delete track points",
+                        required=True)
+    parser.add_argument('-where', dest='where', help="Query conditions for polygons you want to use in your cleanup. Defaults to all (1=1)", default="1=1")
+    parser.add_argument('-delete-inside', dest='inside', help="If true, delete the tracks inside the polygon. If false, delete the tracks outside the polygon", required=True)
```

Comment:
What do you think of the name of this parameter? I don't really like it

Reply:
Hmm what about something like `-mode intersection/symetric-difference` ? Default would be intersection